soletta/flow_node.go

```diff
@@ -0,0 +1,77 @@
+package soletta
+
+/*
+#include <soletta.h>
+#include <sol-flow.h>
+*/
+import "C"
+import "unsafe"
+import "image/color"
+import "time"
+
+//Represents a node in the flow based programming paradigm
+type FlowNode struct {
+	cnode *C.struct_sol_flow_node
+}
+
+//Retrieves the type associated with the flow node
+//Returns a nil value in case of error
+func (fn *FlowNode) GetType() *FlowNodeType {
+	ctype := C.sol_flow_node_get_type(fn.cnode)
+	if ctype == nil {
+		return nil
+	}
+	return &FlowNodeType{ctype}
+}
+
+//Retrieves the port index by its name
+func (fn *FlowNode) GetPort(name string, direction int) (portIndex uint16, ok bool) {
+	t := fn.GetType()
+	if t == nil {
+		return C.UINT16_MAX, false
+	}
+	return t.GetPort(name, direction)
+}
+
+//Sends a packet on this port
+func (fn *FlowNode) SendPacket(packetType string, value interface{}, port uint16) {
```

Comment:
this function should return a value in case of error. `packetType` can be invalid or `C.sol_flow_send_*_packet()` can fail.